### ISSUE TITLE
Fable 4.0.6 breaks Fable.Remoting deserialization

### ISSUE THREAD

**Issue by OnurGumus** [fable-compiler/Fable#3418]
### Description

It is is hard to be a bit specific for my case because code isn't public. But I have some code using Fable.Remoting.  It fetches some data from server via Fable.Remoting and Elmish commands. In 4.0.5 compiler it works fine. When I use 4.0.6 it returns all zeroes without any errors

**Comment by OnurGumus** [comment]
The other side is a giraffe server. So no Fable there. When I say it uses Fable 4.0.6, I refer to the dotnet tool.

The interesting bit is the data I was receving is **int64** and they are coming 0 instead of real value. Reverting 4.0.5 from dotnet tool fixes the problem.

**Comment by ncave** [comment]
@OnurGumus The `int64` change to native JS `bigint` was introduced in 4.0.5.
If it works with 4.0.5, then it's probably something else then.

I don't suppose you can provide a snippet or sample of the server's response json?
What is being used for JSON serialization on the server?

**Comment by ncave** [comment]
@OnurGumus
> I verified server response is correct.

Then it must be the JSON deserialization, if you can provide a snippet of what that `int64` looks like in the response JSON, we'll be able to see what the problem.

**Comment by OnurGumus** [comment]
In my case i tlooks like below
```
{"VisitorId":{"VisitorId":"tb4q8ARZSVCrFucPXchi"},"Credit":{"Credit":"+782"},"Version":{"Version":"+85"}}
```

**Comment by Zaid-Ajaj** [comment]
@OnurGumus the issue might be in Fable.SimpleJson which handles the deserialization business on the client side of things, used by Fable.Remoting.Client.

**Comment by OnurGumus** [comment]
Perhaps, but the fact is ,it works well with all fable but 4.06

**Comment by ncave** [comment]
@OnurGumus Is `Fable.SimpleJson` what's being used for serialization on both ends?

Standard Giraffe JSON serialization produces this for `int64`:
```json
{"text":"0x400000000000000","int64":288230376151711744}
```

**Comment by alfonsogarciacaro** [comment]
Before we had a toJson method in Long.js prototype (which is called if present by JSON.stringify).
https://github.com/fable-compiler/Fable/blob/e45b25a129636efed422b526cb07c4de8062029e/src/fable-library/lib/long.js#L59

Should we add it to BigInt prototype? I think it's not recommended to modify the prototype of native objects, but here it's proposed as solution.
https://stackoverflow.com/a/70315718

**Comment by ncave** [comment]
@alfonsogarciacaro It's already [there](https://github.com/fable-compiler/Fable/blob/main/src/fable-library/BigInt.ts#L15).

**Comment by OnurGumus** [comment]
I have reconfirmed, it works with 4.0.5 and gets broken with 4.0.6. It uses Fable.Simple.Json on the client to deserialize. It actually seems initial deserialization is correct. Then goes back to elmish async and I lose track. See the screenshot

<img width="659" alt="image" src="https://user-images.githubusercontent.com/4114592/231684135-be58e5a4-8251-4cbf-b09a-5c2f76ff9dc9.png">

**Comment by kerams** [comment]
I'm on 4.0.6 and at first glance both JSON and MsgPack serialization via Remoting looks OK.

**Comment by OnurGumus** [comment]
Actually I believe this issue has nothing to do with int64. I can see entire data is deserialized to empty.

**Comment by ncave** [comment]
@OnurGumus 
> It actually seems initial deserialization is correct.
 Though at this point it is still not converted to an int yet.

If not the initial deserialization, then what converts those fields to `int64`?

**Comment by OnurGumus** [comment]
That’s something @Zaid-Ajaj can answer perhaps.

**Comment by alfonsogarciacaro** [comment]
Thanks a lot you all for the investigation! I think I know where the problem is. @OnurGumus Is the parsed type (or one of the nested values) a **single-case** union?

**Comment by OnurGumus** [comment]
@alfonsogarciacaro  Yes. To be preciese it looks as below
```fsharp
type VisitorId = VisitorId of string

type Credit = Credit of int64
type Version = Version of int64

type CreditData = { VisitorId: VisitorId; Credit: Credit; Version : Version }

```

**Comment by alfonsogarciacaro** [comment]
@OnurGumus Can you please try with Fable 4.1.0-beta-001?

**Comment by OnurGumus** [comment]
@alfonsogarciacaro  problem solved with 4.1.0-beta-001. Thanks for excellent support!
